sematic/db/tests/test_queries.py

```diff
@@ -101,6 +103,43 @@ def test_get_artifact(test_db, persisted_artifact: Artifact):  # noqa: F811
     assert artifact.json_summary == artifact.json_summary
 
 
+def test_save_artifact(test_db, test_storage):  # noqa: F811
+    artifact = make_artifact(42, int, storage=test_storage)
+    save_graph(artifacts=[artifact], runs=[], edges=[])
+    persisted_artifact = get_artifact(artifact.id)  # noqa: F811
+
+    assert persisted_artifact.id == artifact.id
+    assert persisted_artifact.type_serialization == artifact.type_serialization
+    assert persisted_artifact.json_summary == artifact.json_summary
+    assert persisted_artifact.created_at == artifact.created_at
+    assert persisted_artifact.updated_at == artifact.updated_at
+
+
+def test_update_artifact(test_db, test_storage):  # noqa: F811
+    original_artifact = make_artifact(42, int, storage=test_storage)
+    # create copies of these values, as sqlalchemy updates models in-place
+    original_created_at = original_artifact.created_at
+    original_updated_at = original_artifact.updated_at
+    save_graph(artifacts=[original_artifact], runs=[], edges=[])
```

Comment:
The new `_save_artifact` function requires a `Session` parameter, and all the tests in this file only validate functions which create their own sessions, so I maintained the same usage pattern by employing the `save_graph` function here.